File: frontend/lib/models/user_details/stress_level.dart
enum StressLevel {
  low(0, 'low'),
  medium(1, 'medium'),
  high(2, 'high'),
  extreme(3, 'extreme');

  final int value;
  final String nameStr;

  const StressLevel(this.value, this.nameStr);

  String toJson() => nameStr;

  static StressLevel fromJson(String value) {
    return StressLevel.values.firstWhere(
      (e) => e.nameStr == value.toUpperCase(),
      orElse: () => throw ArgumentError('Unknown stress level: $value'),
    );
  }

  int toInt() => value;

  static StressLevel fromInt(int value) {
    return StressLevel.values.firstWhere(
      (e) => e.value == value,
      orElse: () => throw ArgumentError('Invalid stress level value: $value'),
    );
  }
}

File: backend/user_details/enums/stress_level.py
from enum import Enum


class StressLevel(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    EXTREME = "extreme"

    def get_stress_factor(self):
        return {
            StressLevel.LOW: 0.0,
            StressLevel.MEDIUM: 0.03,
            StressLevel.HIGH: 0.07,
            StressLevel.EXTREME: 0.10,
        }[self]
